onnxruntime/core/providers/trt/trt_kernel.cc

```diff
@@ -0,0 +1,259 @@
+
+        ORT_ENFORCE(graph.Resolve().IsOK());
+    }
+    else if (node.NodeType() == onnxruntime::Node::Type::Fused)
+    {
+        //This is a fused node. Refer to a function (sub-graph)
+        const Graph& graph_body = node.GetFunctionBody()->Body();
+        for (auto& graph_body_node : graph_body.Nodes())
+        {
+            graph.AddNode(graph_body_node);
+        }
+        ORT_ENFORCE(graph.Resolve().IsOK());
+
+        //Add inputs to graph
+        for (int i = 0, end = node.InputDefs().size(); i < end; ++i)
+        {
+            const onnxruntime::Tensor* temp = nullptr;
+            const onnxruntime::Tensor** constant_input_value = &temp;
+            auto stat = info.TryGetConstantInput(i, constant_input_value);
+            if (stat) //initialized inputs
+            {
+                auto& input_tensor = *constant_input_value;
+                const auto& tensor_type = input_tensor->DataType();
+                TensorProto::DataType dtype = TensorProto_DataType_UNDEFINED;
+                TensorProto tensor_proto;
+                if (tensor_type == DataTypeImpl::GetType<float>())
+                {
+                    dtype = TensorProto_DataType_FLOAT;
+                    const auto& input_data = input_tensor->Data<float>();
+                    for (int j = 0, end = input_tensor->Shape().Size(); j < end; ++j)
+                    {
+                        tensor_proto.add_float_data(input_data[j]);
+                    }
+                }
+                else if (tensor_type == DataTypeImpl::GetType<int64_t>())
+                {
+                    dtype = TensorProto_DataType_INT64;
+                    const auto& input_data = input_tensor->Data<int64_t>();
+                    for (int j = 0, end = input_tensor->Shape().Size(); j < end; ++j)
+                    {
+                        tensor_proto.add_int64_data(input_data[j]);
+                    }
+                }
+                else  //TODO: add other tensor types
+                {
+                    dtype = TensorProto_DataType_UNDEFINED;
+                }
+                tensor_proto.set_name(node.InputDefs()[i]->Name());
+                auto& shape = input_tensor->Shape().GetDims();
+                for (auto& dim : shape)
+                {
+                    tensor_proto.add_dims(dim);
+                }
+                tensor_proto.set_data_type(dtype);
+                graph.AddInitializedTensor(tensor_proto);
+            }
+            else  //graph inputs
+            {
+                graph_input_index_.push_back(i);
+                graph_input_name_.push_back(node.InputDefs()[i]->Name());
+            }
+        }
+    }
+
+    ONNX_NAMESPACE::ModelProto model_p = model.ToProto();
+
+    //Add node's outputs to graphproto's outputs if the node's EdgeEnd nodes are not in the graph
+    std::set<string> output_set;
+    for (int i = 0, end = model_p.graph().output().size(); i < end; ++i)
```

Comment:
>model_p.graph().outpu [](start = 26, length = 21)

Whilst all the function calls should get optimized out, they may not for various reasons. To be sure, could we instead have a local reference to the values that are being used repeatedly?

e.g.
model_p.graph()
model_p.graph().output()
model_p.graph().value_info()
model_p.mutable_graph()->mutable_output()